pkg/scheduler/apis/config/validation/validation_test.go

```diff
@@ -475,6 +476,46 @@ func TestValidateKubeSchedulerConfigurationV1beta3(t *testing.T) {
 	goodRemovedPlugins2 := validConfig.DeepCopy()
 	goodRemovedPlugins2.Profiles[0].Plugins.Score.Enabled = append(goodRemovedPlugins2.Profiles[0].Plugins.Score.Enabled, config.Plugin{Name: "PodTopologySpread", Weight: 2})
 
+	invalidPercentNumOfNodesToScore := validConfig.DeepCopy()
+	invalidPercentNumOfNodesToScoreVal := intstr.FromString("invalid")
+	invalidPercentNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &invalidPercentNumOfNodesToScoreVal
+
+	negativePercentNumOfNodesToScore := validConfig.DeepCopy()
+	negativePercentNumOfNodesToScoreVal := intstr.FromString("-10%")
+	negativePercentNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &negativePercentNumOfNodesToScoreVal
+
+	zeroPercentNumOfNodesToScore := validConfig.DeepCopy()
+	zeroPercentNumOfNodesToScoreVal := intstr.FromString("0%")
+	zeroPercentNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &zeroPercentNumOfNodesToScoreVal
+
+	normalPercentNumOfNodesToScore := validConfig.DeepCopy()
+	normalPercentNumOfNodesToScoreVal := intstr.FromString("30%")
+	normalPercentNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &normalPercentNumOfNodesToScoreVal
+
+	decimalPercentNumOfNodesToScore := validConfig.DeepCopy()
+	decimalPercentNumOfNodesToScoreVal := intstr.FromString("33.33%")
+	decimalPercentNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &decimalPercentNumOfNodesToScoreVal
+
+	percentNumOfNodesToScore100 := validConfig.DeepCopy()
+	percentNumOfNodesToScore100Val := intstr.FromString("100%")
+	percentNumOfNodesToScore100.Profiles[0].NumOfNodesToScore = &percentNumOfNodesToScore100Val
+
+	percentNumOfNodesToScore150 := validConfig.DeepCopy()
+	percentNumOfNodesToScore150Val := intstr.FromString("150%")
+	percentNumOfNodesToScore150.Profiles[0].NumOfNodesToScore = &percentNumOfNodesToScore150Val
+
+	negativeIntNumOfNodesToScore := validConfig.DeepCopy()
+	negativeIntNumOfNodesToScoreVal := intstr.FromInt(-10)
+	negativeIntNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &negativeIntNumOfNodesToScoreVal
+
+	zeroIntNumOfNodesToScore := validConfig.DeepCopy()
+	zeroIntNumOfNodesToScoreVal := intstr.FromInt(0)
+	zeroIntNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &zeroIntNumOfNodesToScoreVal
+
+	normalIntNumOfNodesToScore := validConfig.DeepCopy()
+	normalIntNumOfNodesToScoreVal := intstr.FromInt(50)
+	normalIntNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &normalIntNumOfNodesToScoreVal
```

Comment:
nit: shall we abstract out a function to make the code more concise?

Reply:
Done